lib/util/util.js

```diff
@@ -0,0 +1,29 @@
+const { inspect } = require('util');
+
+const isBlank = (x) => (x === null) || (x === undefined) || (x === '');
+
+const merge = (...objs) => Object.assign({}, ...objs);
+const without = (obj, ...reject) => {
+  const result = Object.assign({}, obj);
+  for (const key of reject)
+    delete result[key];
+  return result;
+};
+
+const printPairs = (obj) => {
+  const result = [];
+  for (const key in obj)
+    result.push(`${key}: ${inspect(obj[key])}`);
+  return result.join(', ');
+};
+
+const ensureArray = (x) => Array.isArray(x) ? x : [ x ];
+
+const reject = Promise.reject.bind(Promise);
+const resolve = Promise.resolve.bind(Promise);
+
+const Trait = (classFunc) => (superclass) => (context) => classFunc(superclass(context), context);
+const Subclass = (superclass, classFunc) => (context) => classFunc(superclass(context), context);
```

Comment:
Is this a common pattern? I have a hard time wrapping my head around it and it makes all class definitions look surprising.

Reply:
No, it's goofy because of dependency injection. I'm redoing it a bit.